src/html-pipe.js

```diff
@@ -98,6 +98,9 @@ export async function htmlPipe(state, req) {
       const level = res.status >= 500 ? 'error' : 'info';
       log[level](`pipeline status: ${res.status} ${res.error}`);
       res.headers.set('x-error', cleanupHeaderValue(res.error));
+      if (res.status <= 400) {
```

Comment:
why `<= 400`  and not `< 400` or `< 500` ?

Reply:
I would think we should return the response headers for a 404 as well. So feels like it should be `< 500`.

Reply:
```suggestion
      if (res.status < 500) {
```